### ISSUE TITLE
Click on Ask's blue done button gets sent through to whatever else is on stage (and with wrong scaling & offset)

### ISSUE THREAD

**Issue by TheLogFather** [LLK/scratch-flash#600]
See http://scratch.mit.edu/discuss/post/736072/ (and note the comment in the project I reference in that post, as well as the project referenced in the OP of that topic).

The click gets sent through to whatever sprite/clone lies at the point of the click's co-ordinates (but with the wrong offset and scaling, so it can appear to be at a completely different place if in fullscreen).

Am I correct in thinking the solution is just to prevent propagation of the mousedown event if the hitTestPoint is satisfied at line 58 of AskPrompter.as...?

**Comment by 2jour** [comment]
@TheLogFather if i use this project : http://scratch.mit.edu/projects/46487418 

1. Click on the done button for ask 

The boy at the bottom right says hello. This is the glitch because the mouse click is also affecting the boy sprite, correct?

**Comment by TheLogFather** [comment]
Yes, the click is going through to the boy. Even more curious is that the boy does not have to be actually behind the button itself - it can be a bit above it (see the position of the 'cancel' button in the original project).

However, exactly where the click will go through to depends on your browser view settings and the mode you're playing the project.

If you have your browser showing the page at normal size (i.e. hit cmd zero on Safari or FF on Mac) then it will be just above the blue button (I think it's offset by the height of the bar above the player). If you have the browser showing the page scaled up (e.g. hit cmd plus on Safari or FF), then the click goes through offset to the left somewhat. If you're playing in fullscreen then the click is offset by a long day above and left.

Basically, it looks like the click ends up at an absolute player-window pixel location, rather than scaled and offset correctly for the player size.

A good demonstration is here: http://scratch.mit.edu/projects/44924892/
But to show it in the normal player case, and with standard browser view scaling, you have to edit the project to make the cats go right to the far right-hand side of the player (add an extra three or so across).

Having said all of that, I suspect that just preventing the click from propagating will mean the whole scaling/offset issue is irrelevant...

Hope that makes sense!

**Comment by Paddle2See** [comment]
Spent quite awhile messing with a project because of this weird behavior.  I documented it again here, if you need another test case

https://scratch.mit.edu/projects/78002238/#editor

**Comment by TheLogFather** [comment]
Made a project that helps to show how the 'click-through' position is dependent upon the state of the project window: https://scratch.mit.edu/projects/86263314/

Try it with browser window set to different zoom, and in fullscreen, and you should get the idea...
